### PR TITLE
[Search] Booster improvements

### PR DESCRIPTION
**Pull request by SferaDev**
Requires:

- https://github.com/xataio/xata/pull/1531
- https://github.com/xataio/xata/pull/1529

### CODE REVIEW

**Comment by SferaDev** [comment]
Pinging @tsg to take a look

**Review by gimenete — APPROVED** [review]

**Comment by SferaDev** [comment]
Holding this until backend is merged

**Pull request merged**
